### PR TITLE
WL 21447 - Create a Skybox Changer

### PR DESCRIPTION
**Pull request by cain-kilgore**
## Test Plan
1. Run this Script: http://mpassets.highfidelity.com/05904016-8f7d-4dfc-88e1-2bf9ba3fac20-v1/skyboxchanger.js
2. Confirm that it launches correctly and you see the Skybox Changer Interface in Desktop & Tablet Mode.
3. Choose each Skybox and confirm that after a few seconds, the Skybox you are standing in changes.
4. Confirm that you can't modify a locked Skybox.
5. Confirm that you need to have Rez Permission to modify the Skybox.

per [Worklist #21447](https://worklist.net/21447)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2462/HighFidelity-Beta-PR11181-8c9f3ae9775d15101b2b6f966f6a4195c651dc58.dmg

**Comment by cain-kilgore** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2464/HighFidelity-Beta-PR11181-8c9f3ae9775d15101b2b6f966f6a4195c651dc58.dmg

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2465/HighFidelity-Beta-PR11181-8c9f3ae9775d15101b2b6f966f6a4195c651dc58.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2465/HighFidelity-Beta-PR11181-8c9f3ae9775d15101b2b6f966f6a4195c651dc58.exe

**Comment by Nex-Pro** [comment]
Testing this

**Comment by Nex-Pro** [comment]
### Pass

### Feedback: 
@themelissabrown 

**"You can set the current Skybox of the Zone you're standing in by clicking on any of the 6 pre-determined skyboxes below."**  

Too wordy.
Might not be standing. 
'6' is hard coded. Should be "six" if at all.
 
**Suggestions:**

"Click an image to Change the Skybox you are in."
"Choose a new Skybox look."
"Click an image to choose a new Skybox."

**Comment by cain-kilgore** [comment]
Initially I showed @misslivirose this and she was happy with it. Might be best asking her to chime in. I do agree with the 6 though, that's my bad.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2469/HighFidelity-Beta-PR11181-185c8da491bb790be50af33aba73d388b9587264.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2469/HighFidelity-Beta-PR11181-185c8da491bb790be50af33aba73d388b9587264.exe

**Comment by misslivirose** [comment]
@cain-kilgore I like the idea of shortening it to "Click an image to choose a new skybox" 
Also, as I'm thinking about it, this should probably also change the ambientURL property as well as the url.

**Comment by cain-kilgore** [comment]
@misslivirose I'll get that changed asap - in regards to the Ambient URL, what would you like me to use?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2470/HighFidelity-Beta-PR11181-619da8c4154cca2c232cd562d2a76569f5a128aa.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2470/HighFidelity-Beta-PR11181-619da8c4154cca2c232cd562d2a76569f5a128aa.exe

File: interface/resources/qml/hifi/SkyboxChanger.qml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+import QtQuick 2.5
+import QtQuick.Controls 1.5 as QQControls
+import QtQuick.Layouts 1.3
+import QtQuick.Controls.Styles 1.4
+
+import QtWebEngine 1.2
```
Comment (zfox23) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: Is this necessary?

File: interface/resources/qml/hifi/SkyboxChanger.qml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import QtWebChannel 1.0
```
Comment (zfox23) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: Is this necessary?

File: interface/resources/qml/hifi/SkyboxChanger.qml [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+import "../styles-uit"
+import "../controls-uit" as HifiControls
+import "../windows"
```
Comment (zfox23) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: Is this necessary?

File: interface/resources/qml/hifi/SkyboxChanger.qml [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+import QtQuick 2.5
+import QtQuick.Controls 1.5 as QQControls
+import QtQuick.Layouts 1.3
+import QtQuick.Controls.Styles 1.4
+
+import QtWebEngine 1.2
+import QtWebChannel 1.0
+
+import "../styles-uit"
+import "../controls-uit" as HifiControls
```
Comment (zfox23) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: Is this necessary?

File: interface/resources/qml/hifi/SkyboxChanger.qml [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+import "../windows"
+import "../controls"
```
Comment (zfox23) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: Is this necessary?

File: interface/resources/qml/hifi/SkyboxChanger.qml [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+
+Rectangle {
+    id: root;
+
+    HifiConstants { id: hifi; }
+
+    property string title: "";
```
Comment (zfox23) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: What's this for?

File: interface/resources/qml/hifi/SkyboxChanger.qml [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+        }
+        RalewaySemiBold {
+            id: titleBarDesc;
+            text: "Click an image to choose a new Skybox.";
+            wrapMode: Text.Wrap
+            // Text size
+            size: 14;
+            // Anchors
+            anchors.fill: parent;
+            anchors.topMargin: 56;
```
Comment (zfox23) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: This goes beyond the height of its container.

File: interface/resources/qml/hifi/SkyboxChanger.qml [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+            anchors.leftMargin: 16;
+            anchors.rightMargin: 16;
+            // Style
+            color: hifi.colors.lightGrayText;
+            // Alignment
+            horizontalAlignment: Text.AlignHCenter;
+            verticalAlignment: Text.AlignVCenter;
+        }        
+    }
+    RowLayout {
```
Comment (zfox23) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: Using `RowLayout`s instead of a View with a Model is not sustainable; it'll be harder than it has to be to add new Skyboxes to this app. Is that OK?

Comment (zfox23) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: Also, why aren't you using a [Grid](http://doc.qt.io/qt-5/qml-qtquick-grid.html)?

File: interface/resources/qml/hifi/SkyboxChanger.qml (hunk 1 above)
Comment (cain-kilgore) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: This has been updated now in the new commit.

File: interface/resources/qml/hifi/SkyboxChanger.qml (hunk 2 above)
Comment (cain-kilgore) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: This has been updated now in the new commit.

File: interface/resources/qml/hifi/SkyboxChanger.qml (hunk 3 above)
Comment (cain-kilgore) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: This has been updated now in the new commit.

File: interface/resources/qml/hifi/SkyboxChanger.qml (hunk 4 above)
Comment (cain-kilgore) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: This has been updated now in the new commit.

File: interface/resources/qml/hifi/SkyboxChanger.qml (hunk 5 above)
Comment (cain-kilgore) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: This has been updated now in the new commit.

File: interface/resources/qml/hifi/SkyboxChanger.qml (hunk 6 above)
Comment (cain-kilgore) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: This has been updated now in the new commit.

File: interface/resources/qml/hifi/SkyboxChanger.qml (hunk 7 above)
Comment (cain-kilgore) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: This has been changed to take the margin from the title instead of a hardcoded value.

File: interface/resources/qml/hifi/SkyboxChanger.qml (hunk 8 above)
Comment (cain-kilgore) [inline interface/resources/qml/hifi/SkyboxChanger.qml]: I've added a comment about this - I do agree, and it's something that should be implemented at some point in the future if there were to be more Skyboxes to be added. In this case as my Worklist task only specified 6, this should be sustainable for now.

**Comment by zfox23** [comment]
NOICE. Looks good to me :+1: Merging!

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2486/HighFidelity-Beta-PR11181-e7a4bff705a8764ab72bbc4edb37a009a664ee4c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2486/HighFidelity-Beta-PR11181-e7a4bff705a8764ab72bbc4edb37a009a664ee4c.exe
